### ISSUE TITLE
MatchError in AsyncWriteJournal

### ISSUE THREAD

**Issue by WellingR** [akka/akka#18303]
Using akka-persistence_2.11-2.4.0-RC1
I see some MatchErrors occur in the log:

    scala.MatchError: Success(DeleteMessagesSuccess(1)) (of class scala.util.Success)
        at akka.persistence.journal.AsyncWriteJournal$$anonfun$1$$anonfun$applyOrElse$11.apply(AsyncWriteJournal.scala:156)
        at akka.persistence.journal.AsyncWriteJournal$$anonfun$1$$anonfun$applyOrElse$11.apply(AsyncWriteJournal.scala:156)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
        at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
        at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:90)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:39)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

It appears to be caused by the following code 
```scala
    case d @ DeleteMessagesTo(persistenceId, toSequenceNr, persistentActor) ⇒
      breaker.withCircuitBreaker(asyncDeleteMessagesTo(persistenceId, toSequenceNr)) map {
        case _ ⇒ DeleteMessagesSuccess(toSequenceNr)
      } recover {
        case e ⇒ DeleteMessagesFailure(e, toSequenceNr)
      } pipeTo persistentActor onComplete {
        case _ if publish ⇒ context.system.eventStream.publish(d)
      }
```
In the last call to onComplete, the function `case _ if publish => ...` causes a MatchError to be thrown when `publish` is false

**Comment by ktoso** [comment]
Ouch, great catch @WellingR - thanks for reporting. 
I'll fix that right away.

**Comment by ktoso** [comment]
Grepped the code base for instances of this mistake, thankfully it's only present in 2 places. 
To be fixed via #18305, thanks for reporting!

**Comment by patriknw** [comment]
if this is a blocker, a workaround is that you define config property:

    akka.persistence.publish-plugin-commands = on
